drivers/volume/portworx/portworx.go

```diff
@@ -1403,10 +1421,8 @@ func (d *portworx) ValidateCreateVolume(volumeName string, params map[string]str
 	if vol.Spec.ProxySpec != nil && vol.Spec.ProxySpec.ProxyProtocol == api.ProxyProtocol_PROXY_PROTOCOL_PURE_BLOCK {
 		// Checking the device path when state is attached
 		if vol.State == api.VolumeState_VOLUME_STATE_ATTACHED && !strings.Contains(vol.DevicePath, DeviceMapper) {
-			return &ErrFailedToInspectVolume{
-				ID:    volumeName,
-				Cause: fmt.Sprintf("Failed to validate device path [%s]", vol.DevicePath),
-			}
+			_, err := task.DoRetryWithTimeout(volumeInspectDevicePath, 5*time.Minute, inspectVolumeRetryInterval)
+			return fmt.Errorf("Failed to validate the device path for a volume [%s] : %v", volumeName, err)
```

Comment:
is there a reason to change the way we are returning? Can we retain `return &ErrFailedToInspectVolume`

Reply:
yes sure, we can do that ,i have reverted that